pkg/controller/machine/controller.go

```diff
@@ -194,9 +194,14 @@ func (r *ReconcileMachine) Reconcile(request reconcile.Request) (reconcile.Resul
 }
 
 func (r *ReconcileMachine) getCluster(ctx context.Context, machine *clusterv1.Machine) (*clusterv1.Cluster, error) {
+	if machine.Labels["cluster"] == "" {
+		return nil, errors.New("missing `cluster` label in Machine spec")
```

Comment:
Just a nit here, but labels are not part of the Machine Spec, but rather they are part of the Machine ObjectMeta

Reply:
Since this label based approach is still a relatively weak link between a machine and a cluster, maybe `getCluster()` should return `nil` if the label is not found instead of an error.